database/raft/membership.go

```diff
@@ -0,0 +1,22 @@
+package raft
+
+import "context"
+
+const potentialMemberPrefix = "potential"
```

Comment:
How do you feel about leading `/` in raft keys? I think currently we sometimes do it and sometimes not.

Separately, (nit) I am not sure "potential" is the right word, because as soon as they join they'll be actual members, not potential. But they should still be in this list.

Thirdly, do you think this set of things should be namespaced along with everything else this package does internally into like a `raft/` (or `/raft/`) prefix? So in this case might end up as `raft/potential/example.com:1234`.